pkg/credentialprovider/config.go

```diff
@@ -74,9 +74,9 @@ func GetPreferredDockercfgPath() string {
 	return preferredPath
 }
 
-func ReadDockerConfigFile() (cfg DockerConfig, err error) {
+func ReadDockerConfigFile(firstDockercfgPath string) (cfg DockerConfig, err error) {
 	// Try happy path first - latest config file
-	dockerConfigJsonLocations := []string{GetPreferredDockercfgPath(), workingDirPath, homeJsonDirPath, rootJsonDirPath}
+	dockerConfigJsonLocations := []string{firstDockercfgPath, GetPreferredDockercfgPath(), workingDirPath, homeJsonDirPath, rootJsonDirPath}
```

Comment:
rather than combining an arg and a global var lookup, can we make this take `paths []string`? if an empty list is passed, the default search paths are used.

```
func GetDockercfgPaths() []string {
  return []string{GetPreferredDockercfgPath(), workingDirPath, homeJsonDirPath, rootJsonDirPath}
}

func ReadDockerConfigFile(paths []string) (cfg DockerConfig, err error) {
  if len(paths) == 0 {
    paths = GetDockercfgPaths()
  }
  ...
}
```